bittensor/_subtensor/subtensor_impl.py

```diff
@@ -1490,16 +1491,56 @@ def neurons(self, block: int = None ) -> List[SimpleNamespace]:
             neuron (List[SimpleNamespace]):
                 List of neuron objects.
         """
+        if self.use_fast_sync:
+            try:
+                return self.neurons_fast(block)
+            except FastSyncException as e:
+                logger.warning("Failed to get neurons fast, falling back to manual sync.: {}".format(e))
+                self.use_fast_sync = False
+                return self.neurons(block)
+        
         neurons = []
         for id in tqdm(range(self.get_n( block ))): 
             try:
                 neuron = self.neuron_for_uid(id, block)
                 neurons.append( neuron )
-            except Exception as e:
+            except NeuronPullException as e:
                 logger.error('Exception encountered when pulling neuron {}: {}'.format(id, e))
                 break
         return neurons
 
+    def neurons_fast(self, block: int = None ) -> List[SimpleNamespace]: 
+        r""" Returns a list of neuron from the chain, using the bundled subtensor-node-api
+        Args:
+            block (int):
+                block to sync from.
+        Returns:
+            neuron (List[SimpleNamespace]):
+                List of neuron objects.
+
+        Raises:
+            FastSyncException: If there is an issue during fast sync
+        """
+        neurons = []
+        if block is None:
+            block: int = self.get_current_block()
+
+        block_hash: str = self.substrate.get_block_hash( block )
+        endpoint_url: str = self.chain_endpoint
+        endpoint_url = bittensor.utils.networking.get_formatted_ws_endpoint_url(endpoint_url)
+    
+        # check if fast sync is available
+        FastSync.verify_fast_sync_support()
+        # try to fast sync
+        fast_sync: FastSync = FastSync(endpoint_url)
+        # get neurons
+        fast_sync.sync_neurons(block_hash)
+        # load neurons
+        neurons = fast_sync.load_neurons()
```

Comment:
given that comment is the name of the function i would say that we do not need it